src/main/kotlin/org/dropProject/services/AssignmentService.kt

```diff
@@ -19,28 +19,44 @@
  */
 package org.dropProject.services
 
+import org.dropProject.controllers.ReportController
 import org.dropProject.dao.Assignment
+import org.dropProject.dao.ProjectGroup
+import org.dropProject.data.GroupedProjectGroups
 import org.dropProject.extensions.realName
-import org.dropProject.repository.AssigneeRepository
-import org.dropProject.repository.AssignmentACLRepository
-import org.dropProject.repository.AssignmentRepository
-import org.dropProject.repository.SubmissionRepository
+import org.dropProject.repository.*
 import org.springframework.cache.annotation.Cacheable
 import org.springframework.stereotype.Service
+import org.springframework.ui.ModelMap
 import java.security.Principal
+import javax.servlet.http.HttpServletRequest
 
+/**
+ * AssignmentService provides [Assignment] related functionality (e.g. list of assignments).
+ */
 @Service
 class AssignmentService(
         val assignmentRepository: AssignmentRepository,
         val assignmentACLRepository: AssignmentACLRepository,
         val submissionRepository: SubmissionRepository,
-        val assigneeRepository: AssigneeRepository
+        val assigneeRepository: AssigneeRepository,
+        val submissionService: SubmissionService,
+        val assignmentTestMethodRepository: AssignmentTestMethodRepository
 ) {
 
     @Cacheable(
             value = ["archivedAssignmentsCache"],
             key = "#principal.name",
             condition = "#archived==true")
+
```

Comment:
Please switch the documentation with the annotation (see https://stackoverflow.com/questions/3109950/codestyle-put-javadoc-before-or-after-annotation)

Reply:
Done